app/services/foreman_templates/template_importer.rb

```diff
@@ -154,5 +155,19 @@ def update_job_template(name, text)
       result = "  #{c_or_u} Template #{id_string}:#{name}"
       { :diff => diff, :status => template.save, :result => result }
     end
+
+    def purge!
+      # TODO add filter on template 'kind'
+      # kind is an ID, need to map / match with lookup table
+      #clause = [filter, kind].join(" AND ")
+      # TODO: Make query more flexible, robust and secure from SQL injection
+      match  = @negate ? 'NOT LIKE' : 'LIKE'
+      clause = @filter || "name #{match} '#{@prefix}%'"
```

Comment:
Are we expecting the user to supply `@filter` with a value like `name LIKE '%foobar%'`? This looks like it needs some SQL handling if so.